### PR TITLE
Hack 19: Submit checkboxes, radios, and select as text fields in Mail…

### PR DESCRIPTION
**Pull request by brianndiiri**
…chimp

### CODE REVIEW

File: webflow-hacks/form-hacks/19/19.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    !formCheckboxes ||
+    !formSelect ||
+    !formRadios
+  )
+    return;
+
+  // on submit form
+  form.addEventListener('submit', function (e) {
+    // prevent default behavior
+    e.preventDefault();
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/19/19.ts]: You have to call `e.stopPropagation()` if you want to prevent a Webflow form submission

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]
Most of this code is unnecessary.
- Select inputs are already sent as text natively.
- Radios are already sent as text natively. It's sent as: `Radio Group Name: Chosen Radio Value`

Only the checkboxes are necessary to be converted from `true | false` to a specific string!

**Comment by alexiglesias93** [comment]
@brayanfinsweet I'm having a hard time understanding this new code. Would you mind sharing the website where you're testing it?

**Comment by brianndiiri** [comment]
https://finsweetwebflowhacks-rewrite.webflow.io/19
They say Mailchimp doesn't accept anything other than text/hidden fields on the form action, so the code populates radio and checkbox data to hidden text fields. It seems though that the form action link is no longer working so there is no way to see this in action

**Comment by alexiglesias93** [comment]
@brayanfinsweet could you create a free Mailchimp account to test it?

**Comment by brianndiiri** [comment]
Hi mailchimp has since added these fields to their interface so this solution is outdated
Screenshot from 2022-08-18 12-40-08

**Comment by alexiglesias93** [comment]
> Hi mailchimp has since added these fields to their interface so this solution is outdated Screenshot from 2022-08-18 12-40-08

@brayanfinsweet so this means that no JS is required for submitting checkboxes and radio buttons to Mailchimp, is that correct?
In that case, we can remove this Hack

**Comment by brianndiiri** [comment]
Correct! this pull request will be closed here

**Pull request closed without merging**